docs/content/Dialog2.mdx

```diff
@@ -0,0 +1,112 @@
+---
+title: Dialog v2
+---
+
+import State from '../components/State'
+
+The dialog component the Primer implementation of the ARIA design pattern [Dialog](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal). A dialog is a type of overlay that can be used for confirming actions, asking for disambiguation, and presenting small forms. They generally allow the user to focus on a quick task without having to navigate to a different page.
+
+**Dialogs appear in the page after a direct user interaction**. Don't show dialogs on page load or as system alerts.
+
+**Dialogs appear centered in the page**, with a visible backdrop that dims the rest of the window for focus.
+
+**All dialogs should have a title and a close button**. Use an overlay header to compose these required elements.
+
+**Dialogs are modal**. Dialogs can be dismissed by clicking on the close button, or by interacting with another button in the overlay. To avoid losing information and missing important messages, clicking outside of the dialog will not close it.
+
+**Make sure larger dialogs remain mobile-friendly**. Even large dialogs need to be responsive. A dialog's width and height will be readjusted depending on the screen size and should never exceed the available space. Use responsive solutions to adjust the UI so they behave well on smaller screens.
+
+**Simple and small dialogs can remain as-is on mobile**. As more elements are added to it, mobile-specific style variations such as **Bottom sheet** and **Full-screen** should be considered.
+
```

Comment:
Should the text inside dialogs be selectable? I can't select text with the current implementation:

https://user-images.githubusercontent.com/4608155/115301851-9cfb4f00-a116-11eb-8720-8894f600efda.mp4

Reply:
This is super weird! It's only happening on the docs site and only on Chrome & Safari. I wonder if something is eating the click events? I might punt on fixing this for now if I don't find the cause soon.

Reply:
Opened #1178 to track this.